tasks/analysis.py

```diff
@@ -603,27 +620,22 @@ def process_impact_report(analysis, report_metadata):
     """
     success = False
     try:
-        # extract report (map and table report)
-        map_reports = [
-            'inasafe-map-report-portrait',  # pdf product
-            'inasafe-map-report-landscape',  # pdf product
-        ]
-        table_reports = [
-            'impact-report',  # html product
-            'impact-report-pdf'  # pdf product
-        ]
-
         # upload using document upload form post request
         # TODO: find out how to upload document using post request
 
-        # assign report to analysis model
-        if os.path.exists(report_metadata['pdf_product_tag'][map_reports[0]]):
-            analysis.assign_report_map(
-                report_metadata['pdf_product_tag'][map_reports[0]])
-        if os.path.exists(
-                report_metadata['pdf_product_tag'][table_reports[1]]):
-            analysis.assign_report_table(
-                report_metadata['pdf_product_tag'][table_reports[1]])
+        for key in report_metadata['pdf_product_tag'].keys():
+            map_report_exists = (
+                'map-report' in key and os.path.exists(
+                    report_metadata['pdf_product_tag'][key]))
+            table_report_exists = (
+                'impact-report-pdf' in key and os.path.exists(
+                    report_metadata['pdf_product_tag'][key]))
+            if map_report_exists:
+                analysis.assign_report_map(
+                    report_metadata['pdf_product_tag'][key])
+            if table_report_exists:
+                analysis.assign_report_table(
+                    report_metadata['pdf_product_tag'][key])
```

Comment:
Nice!
You could do something like this:
```suggestion
        assgin_report = {
            'map-report': analysis.assign_report_map,
            'impact-report-pdf': analysis.assign_report_table
        }
        for key in report_metadata['pdf_product_tag'].keys():
            report_exists = (
                key in assign_report and os.path.exists(
                    report_metadata['pdf_product_tag'][key]))
            if report_exists:
                assign_report[key](report_metadata['pdf_product_tag'][key])
```